variables.tf

```
@@ -1,103 +1,76 @@
-variable "ipam" {
-  description = "Object of inputs for IPAM resources"
+variable "vpc_ipam" {
+  description = "Object of inputs for all IPAM configuration resources"
   type = object({
-    region_name = optional(string)
-    tags        = optional(map(string))
-    cascade     = optional(bool)
+    ipam = optional(object({
+      operating_regions = list(object({
+        region_name = optional(string) #required
+      }))
+      tags        = optional(map(string))
+      cascade     = optional(bool)
+      description = optional(string)
+    }))
+    pool = optional(list(object({
```

When plural, prefer a plural attribute...

```suggestion
    pools = optional(list(object({
```